### ISSUE TITLE
Support room upgrades

### ISSUE THREAD

**Issue by KitsuneRal** [QMatrixClient/libqmatrixclient#235]
The library support of upgrades follows the [room upgrading guide aka MSC1501](https://github.com/matrix-org/matrix-doc/blob/rav/room_upgrades/proposals/1501-room-version-upgrades.md) and consists of the following:
- [x] Support `m.room.tombstone` events (i.e. add them to the event type system so that they get parsed)
- [x] Support reading the predecessor room from `m.room.create` events (depends on #234) 
- [x] Add `successor` and `predecessor` properties to `Room`; non-empty `successor` shall mean that the room has got upgraded and that clients should hide it from the room list and disallow sending messages to it (relates to #213); the library should also actively resist sending messages to rooms with successors (as in: `Room` must disallow it but one can still successfully `callApi<SendEventJob>` in that room)
- [ ] ~~(optional, probably will be tracked separately) Provide API to the clients to retrieve the merged timeline across upgrades, optimising away power level event replicas etc. (depends on #36)~~ (tracked in #280)

Processing upgrades that come through sync does not necessarily depend on #233: the upgrade sequence only mentions room versions in the context of initiating upgrade locally, and this part (`Room::upgrade()`) will be tracked in a separate issue.

**Comment by KitsuneRal** [comment]
Moving out the remainder to #280, and this one is fixed!
